### PR TITLE
Ability to specify multiple paths for migrations and seeds (0.6.x-dev branch)

### PR DESCRIPTION
**Pull request by bartosz-maciaszek**
Rebased and squashed version of [this pull request](https://github.com/robmorgan/phinx/pull/871).

### CODE REVIEW

**Comment by rquadling** [comment]
A couple of points.

1. Should/could the migration/seed paths be displayed realpath()'d
    ```
    Phinx by Rob Morgan - https://phinx.org. version 0.6.1
    
    using config file .\phinx.yml
    using config parser yaml
    using migration paths
     - D:\comptest/db/migrations
     - D:\comptest/db/migrations/first
     - D:\comptest/db/migrations/second
    using seed paths
     - D:\comptest/db/seeds
    ```

2. What thoughts do you have over the status including the path of the migration?
    ```
     Status  Migration ID    Started              Finished             Migration Name
    ----------------------------------------------------------------------------------
         up  20160621121032  2016-06-21 16:05:53  2016-06-21 16:05:53  First
             BREAKPOINT SET
       down  20160621150534                                            Second
       down  20160621151145                                            Third
    ```
    One of the causes with up'd migrations being marked as missing is when switching branches.
    Being able to incorporate runtime data with the migration (for me it would be git branch and migration path), then the status would be VERY explicit on where to look for the missing migration.

    I think though, that the storing the path wouldn't warrant enough energy.

3. The use of braces needs to be detected and rejected if not possible.
 ```yaml
 paths:
    migrations:
        - %%PHINX_CONFIG_DIR%%/db/migrations
        - %%PHINX_CONFIG_DIR%%/db/migrations/{first,second}
    seeds: %%PHINX_CONFIG_DIR%%/db/seeds
 ```
 ```
 [ 21/06/2016 16:21:26 ] [ D:\comptest ] >vendor\bin\phinx test
 Phinx by Rob Morgan - https://phinx.org. version 0.6.1

 using config file .\phinx.yml
 using config parser yaml


   [InvalidArgumentException]
   Migration directory "D:\comptest/db/migrations/{first,second}" does not exist


 test [-c|--configuration CONFIGURATION] [-p|--parser PARSER] [-e|--environment ENVIRONMENT]
 ```

 The `test` command needs to also validate multiple paths.


Only number 3 is a stopper at this point.

If you can add some tests to verify that ``test`` is working correctly, verifying the multiple globbed paths, then squash and we should be good to go!

Great work!!!

**Comment by bartosz-maciaszek** [comment]
1. What's the point of this in the first place? Do you want to have consistent forward and backslashes? It does work for me when I used `realpath()`, not sure how appveyor would handle it (as it's on Windows).

2. The problem I see is that not everyone uses git and VCSs can change over the time. It would be nice to have it, but not sure if that's worth it.

3. How would you like it to work? The `test` command validates the multiple paths at the moment, but obviously does not expand braces. But even if it did, it wouldn't match the missing directories. Given you have paths configured as follows...

```
paths:
  migrations:
      - %%PHINX_CONFIG_DIR%%/db/migrations
      - %%PHINX_CONFIG_DIR%%/db/migrations/{first,second}
  seeds: %%PHINX_CONFIG_DIR%%/db/seeds
```

...I cannot really check if both `db/migrations/first/` and `db/migrations/second/` exist. `glob('db/migrations/{first,second}', GLOB_BRACE)` returns only _existing_ directories matching the pattern.

In terms if rejecting the braces when not available - do you think checking if a path contains `{` and `}` and throwing an exception if `GLOB_BRACE` is not available would suffice?

**Comment by rquadling** [comment]
1. SOME of the paths are shown realpath()'d (so all the slashes are the right way round based on OS). 4 calls to realpath(). Not really important. Cosmetic.

2. Yeah. I'm thinking the same. A nice idea. WIBNI.

3. The paths do have to exist.

Based upon the following content on my machine ...
```
D:\comptest\db\migrations
D:\comptest\db\migrations\20160621151145_third.php
D:\comptest\db\migrations\first
D:\comptest\db\migrations\first\20160621121032_first.php
D:\comptest\db\migrations\second
D:\comptest\db\migrations\second\20160621150534_second.php
D:\comptest\db\seeds
```

I'm getting output of ...
```
Phinx by Rob Morgan - https://phinx.org. version 0.6.1

using config file .\phinx.yml
using config parser yaml


  [InvalidArgumentException]
  Migration directory "D:\comptest/db/migrations/{first,second}" does not exist


test [-c|--configuration CONFIGURATION] [-p|--parser PARSER] [-e|--environment ENVIRONMENT]
```

The `test` command does need to expand the braces (it doesn't at the moment). The `status`, `migrate` and `rollback`  commands all work fine, but they also do not show the migrations paths appropriately.

```
Phinx by Rob Morgan - https://phinx.org. version 0.6.1

using config file .\phinx.yml
using config parser yaml
using migration paths
 - D:\comptest/db/migrations
 - D:\comptest/db/migrations/{first,second}
using seed paths
 - D:\comptest/db/seeds
warning no environment specified, defaulting to: development

 Status  Migration ID    Started              Finished             Migration Name
----------------------------------------------------------------------------------
     up  20160621121032  2016-06-21 16:05:53  2016-06-21 16:05:53  First
         BREAKPOINT SET
   down  20160621150534                                            Second
   down  20160621151145                                            Third
```

**Comment by bartosz-maciaszek** [comment]
1 & 3 should be fixed now. Also, I think I fixed (at least partially) https://github.com/robmorgan/phinx/issues/833.

Sadly, curly braces complicate things. For example, there's no way to create a migration directory if it does not exist out of the pattern (see `Create` and `SeedCreate` commands). Having multiple paths implemented, I'm wondering if we really need them.

**Comment by rquadling** [comment]
Will be looking at this at the weekend.

**Comment by bartosz-maciaszek** [comment]
Any update on this one?

**Comment by rquadling** [comment]
Planned for this evening. I was waiting on other PRs to finalise, but they aren't ready yet.

**Comment by dancryer** [comment]
Is this likely to be merged any time soon? 

We're still struggling along in our app with the original patch I made for the v0.4.x branch, created in Feb 2015. :(

**Comment by rquadling** [comment]
Oh the life of a developer with a family! Time sucking entities, the lot of them!

**Comment by rquadling** [comment]
Another rebase please.

File: phinx.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 paths:
-    migrations: %%PHINX_CONFIG_DIR%%/db/migrations
+    migrations:
```
Comment (rquadling) [inline phinx.yml]: As this is the like file that is supplied, I don't think it should have any additional paths.

**Comment by rquadling** [comment]
HI. Can you rebase to 0.7.x-dev please. This will be the new breaking changes branch (new interface methods or signature changes).

**Comment by bartosz-maciaszek** [comment]
Since github does not support changing branches in pull requests, I have created [another one](https://github.com/robmorgan/phinx/pull/927). Really hope that's the last one.

**Comment by rquadling** [comment]
Should be. New branch is for breaking changes!

**Pull request closed without merging**